mcp-server/src/analysis.ts

```diff
@@ -0,0 +1,560 @@
+ */
+function detectGrowthPhases(values: Array<{date: string; value: number}>): Array<{phase: string; startDate: string; endDate: string; growth: number}> {
+  if (values.length < 3) return [];
+
+  const phases = [];
+  let currentPhaseStart = 0;
+  let currentPhaseType = 'stable';
+
+  for (let i = 1; i < values.length; i++) {
+    const prevValue = values[i - 1]!.value;
+    const currentValue = values[i]!.value;
+    const growth = currentValue - prevValue;
+    const growthRate = prevValue > 0 ? growth / prevValue : 0;
+
+    let phaseType = 'stable';
+    if (growthRate > 0.05) phaseType = 'growth';
+    else if (growthRate < -0.05) phaseType = 'decline';
+
+    if (phaseType !== currentPhaseType && i - currentPhaseStart >= 2) {
+      const phaseGrowth = values[i - 1]!.value - values[currentPhaseStart]!.value;
+      phases.push({
+        phase: currentPhaseType,
+        startDate: values[currentPhaseStart]!.date,
+        endDate: values[i - 1]!.date,
+        growth: phaseGrowth
+      });
+      currentPhaseStart = i - 1;
+      currentPhaseType = phaseType;
+    }
+  }
+
+  if (values.length - currentPhaseStart >= 2) {
+    const phaseGrowth = values[values.length - 1]!.value - values[currentPhaseStart]!.value;
+    phases.push({
+      phase: currentPhaseType,
+      startDate: values[currentPhaseStart]!.date,
+      endDate: values[values.length - 1]!.date,
+      growth: phaseGrowth
+    });
+  }
+
+  return phases;
+}
+
+/**
+ * Detects seasonality in a time-series dataset.
+ *
+ * @param values - Array of time-series data points.
+ * @returns True if seasonality is detected, false otherwise.
+ */
+function detectSeasonality(values: Array<{date: string; value: number}>): boolean {
+  if (values.length < 12) return false; // Need at least a year of data
+
+  const monthlyAverages: Record<string, number[]> = {};
+
+  values.forEach(item => {
+    const month = item.date.substring(5, 7); // Extract MM from YYYY-MM
+    if (!monthlyAverages[month]) monthlyAverages[month] = [];
+    monthlyAverages[month].push(item.value);
+  });
+
+  const monthlyMeans = Object.entries(monthlyAverages)
+    .map(([month, vals]) => ({
+      month,
+      mean: vals.reduce((sum, val) => sum + val, 0) / vals.length
+    }));
+
+  if (monthlyMeans.length < 6) return false; // Data from multiple months
+
+  const overallMean = monthlyMeans.reduce((sum, m) => sum + m.mean, 0) / monthlyMeans.length;
+  const variance = monthlyMeans.reduce((sum, m) => sum + Math.pow(m.mean - overallMean, 2), 0) / monthlyMeans.length;
+  const coefficientOfVariation = overallMean > 0 ? Math.sqrt(variance) / overallMean : 0;
+
+  // If monthly variations are significant, there might be seasonality
+  return coefficientOfVariation > 0.2;
+}
+
+/**
+ * Generates suggestions for handling specific error messages.
+ * NOTE: The suggestions have been generated with AI, so I'd request @frank-zsy / @birdflyi to help enhance this further.
```

Comment:
At present, it should be sufficient. We can wait for feedback and iterate based on it~